Makefile

```diff
@@ -600,3 +600,21 @@ kind-cluster: ## Create a new kind cluster designed for development with Tilt
 
 go-version: ## Print the go version we use to compile our binaries and images
 	@echo $(GO_VERSION)
+
+## --------------------------------------
+## Documentation and Publishing
+## --------------------------------------
+
+.PHONY: serve
+serve: ## Build the CAPIBM book and serve it locally to validate changes in documentation.
+	make -C docs/book/ serve
+
+## --------------------------------------
+## Update Go Version
+## --------------------------------------
+
+.PHONY: update-go
+update-go: ## Update Go version across files - Usage make update-version VERSION=X.YY.ZZ
```

Comment:
can we somehow call this in the verify flow and ensure files are always updated?

Reply:
Hi @mkumatag, 
I had discussed with @Prajyot-Parab regarding this and have added a new target `verify-go-version` which checks if a patch release is available for the n-1'th minor release (Just the way we follow) and chose not to block PRs when the version check fails.

I have a single commit for this action in a different branch to trigger for all PR, which confirms if we're using the latest patch release.  I shall enable this action for this repository if this seems ok.
The current behavior is that a warning is raised in the workflow regarding the version mismatch.
<img width="937" alt="image" src="https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/assets/110517346/f6acb6a9-0526-49dd-b066-85ba7c77b344">

along with the following output in the shell:
```
go1.20.12 is available on go.dev, the version in Makefile is 1.20.11.
Consider bumping the version, run make update-go VERSION=go1.20.12 to update across all files.
```

The above check is a part of `verify` as well. Since we do not mark the job as failed, the findings related to version remains as a few lines of log in the prow-job.

Reply:
I feel allowing user to decide what version they want to use, we can update the version manually or by some automation like dependatbot will be better idea.